buildpipe/pipeline.py

```diff
@@ -11,15 +11,17 @@
 import collections
 from typing import Dict, List, Tuple, Set, Generator, Callable, NoReturn, Union
 
-import box
 import pytz
 import jsonschema
+import ruamel
 from ruamel import yaml
 
 TAGS = List[Union[str, Tuple[str]]]
 
 
-BOX_CONFIG = dict(frozen_box=True, default_box=True)
+class Hashabledict(dict):
+    def __hash__(self):
```

Comment:
While this dictionary is certainly "hashable", because it implements the `__hash__` method, it is not immutable. Therefore insertions into this dictionary may lead to the value returned by `__hash__` to change.

This is not inline with the assumptions that Python makes around [hashable](https://docs.python.org/3/glossary.html#term-hashable) objects and may lead to strange and unexpected behavior, e.g. when this dictionary is used as a key in a data structure that leverages object hashes (i.e. a dict).

Could we perhaps come up with a better solution? I think we should be able to use regular dictionaries, which we can convert into an immutable, hashable [Mapping](https://docs.python.org/3/library/typing.html?highlight=mapping#typing.Mapping) type right before serializing to YAML.

Reply:
Thanks @RikHeijdens. I removed using a set. LMK what you think.